src/core/CoreSession.jl

```diff
@@ -5,61 +5,146 @@ const PULSE_INTERVAL_MILLIS = 5000
+                     type::Int32,
+                     options::GraknOptions = GraknOptions();
+                     request_timout::Real=6) where {T<:AbstractCoreClient}
+   # try
+        options.session_idle_timeout_millis = PULSE_INTERVAL_MILLIS + 1000
+        #building open_request
+        open_req = SessionRequestBuilder.open_req(
             database, type , copy_to_proto(options, grakn.protocol.Options)
         )
+        # open the session
         startTime = now()
-        res, tes = session_open(client.core_stub.blockingStub, gRPCController(), open_req)
-        if res === nothing
-            throw("no session returned")
-        end
+        grpc_controller = gRPCController(request_timeout=request_timout)
+        req_result, status  = session_open(client.core_stub.blockingStub, grpc_controller, open_req)
+        res_id = grpc_result_or_error(req_result, status, result->result.session_id)
         endTime = now()
 
         database = CoreDatabase(database)
         networkLatencyMillis = (endTime - startTime).value
-        session_id = res.session_id
-        transactions = Array{Union{Nothing,<:AbstractCoreTransaction},1}(nothing,0)
+        session_id =  res_id
+        transactions = Dict{UUID,AbstractCoreTransaction}()
         is_open = true
 
-        result = CoreSession(client, database, session_id, transactions, type, options, is_open, networkLatencyMillis)
+        t = Controller(true, 0.5)
+
+        result = CoreSession(client, database, session_id, transactions, type, ReentrantLock() ,options, is_open, networkLatencyMillis, t, request_timout)
+
+        make_pulse_request(result, t)
+
+        # the following transaction is useless for the construction of the the Session
+        # and is only to initialize the first compilation of the ProtoBuf machinary
+        # underneath the transaction. In the environment here it is possible to control
+        # the tick of the pulse request a close as can be. If somebody find a way to
+        # avoid the long starting time of the transaction this can be removed.
+        trans = nothing
+        try
+            trans = transaction(result, Proto.Transaction_Type[:READ])
+        catch ex
+            @info "First attempt for transaction done and not successful"
+        finally
+        end
+        if trans !== nothing
+            close(trans)
+        end
 
-        @async start_pulse(result, (PULSE_INTERVAL_MILLIS / 1000))
+        t.duration_in_seconds =  (PULSE_INTERVAL_MILLIS / 1000) - 0.5
 
         return result
     # catch ex
     #     throw(GraknClientException("Error construct a CoreSession",ex))
     # end
 end
 
-function start_pulse(session::T, pulse_time::Number) where {T<:AbstractCoreSession}
-    while session.isOpen
-        make_pulse_request(session)
-        sleep(pulse_time - 1)
+"""
+function make_pulse_request(session::T) where {T<:AbstractCoreSession}
+    This function make a pulse request to keep the session alive.
```

Comment:
The doc string formatting is more like this:
```suggestion
    make_pulse_request(session::T) where {T<:AbstractCoreSession}

Make a pulse request to keep the session alive.
```